### ISSUE TITLE
fromAll Persistent Subscription returns the same event multiple times

### ISSUE THREAD

**Issue by StevenBlair123** [EventStore/EventStore#3396]
**Describe the bug**
Persistnt Susbcription to fromAll is running and I add a new event.
I receive this event 4 times:

ProductCreated - **Product-6** - {
  "productId": 6
}
ProductCreated - **$streams** - {
  "productId": 6
}
ProductCreated - **$et-ProductCreated** - {
  "productId": 6
}
ProductCreated - **$ce-Product** - {
  "productId": 6
}

**To Reproduce**
1. Create a Persistent Subscription fromAll
2. Add a new event

**Expected behavior**
I think we should only receice this event once.

**Actual behavior**
Event appears 4 times.

**Config/Logs/Screenshots**
If applicable, please attach your node configuration, logs or any screenshots.

**EventStore details**
 - EventStore server version:

 - Operating system:

 - EventStore client version (if applicable): 21.2.1-alpha.0.28"

**Additional context**

Weirdly, if I delete the persistent subscription, and recreate it, I only get the event once:

ProductCreated - **Product-6** - {
  "productId": 6
}

**Comment by oskardudycz** [comment]
@StevenBlair123, could you confirm if you're not setting the `resolveLinkTos` param set to true? If you do, you may get events multiple times, as it will also get events generated by system projections (e.g. by category). They're emitting the link events that will have the same data as the original one after resolution. You can see looking at the event ids if they're indeed the same events or not.

**Comment by StevenBlair123** [comment]
@oskardudycz  - yeah I had resolveLinkTos set to true.
Ran the test again, and can confirm 4 different events arrived.
Still unsure why only 1 event appears after the persistent subscription and re-create it (with resolveLinkTos set to true)

**Comment by oskardudycz** [comment]
Could you send us that one event that you're getting after recreation? It'd be also great if you could share your persistent subscriptions (and database) config so we could try to reproduce it.

**Comment by StevenBlair123** [comment]
@oskardudycz - the server where we saw the problem is now on 21.01.1 and is return us 4 events after recreation (I think this is expected).
I tried locally on a 21.6 and I can't replicate it (I get an error creating the fromAll persistent subscription.

_25612,18,11:13:02.178,ERR] Error while processing message "EventStore.Core.Messages.ClientMessage+CreatePersistentSubscription" in queued handler '"PersistentSubscriptions"'.
System.ArgumentNullException: Value cannot be null. (Parameter 'value')
   at Grpc.Core.Utils.GrpcPreconditions.CheckNotNull[T](T reference, String paramName)
   at Grpc.Core.Metadata.Entry..ctor(String key, String value)
   at Grpc.Core.Metadata.Add(String key, String value)
   at EventStore.Core.Services.Transport.Grpc.RpcExceptions.PersistentSubscriptionFailed(String streamName, String groupName, String reason) in D:\a\TrainStation\TrainStation\build\oss-eventstore\src\EventStore.Core\Services\Transport\Grpc\RpcExceptions.cs:line 131_

I think the main thing is, we were using an older version of evenstore when this first occured and it seems to be working fine in the latest version.
